sir_complainsalot/delphi_sir_complainsalot/check_source.py

```diff
@@ -156,3 +157,15 @@ def check_source(data_source, meta, params, grace, logger):
                 gap_complaints[row["signal"]].geo_types.append(row["geo_type"])
 
     return list(age_complaints.values()) + list(gap_complaints.values())
+
+
+def _is_retired(row, retired_signals):
+    """Determine if a row of the metadata belongs to a retired signal according to the params."""
+    if not retired_signals:
+        return False
+    for signal in retired_signals:
+        if row["signal"] == signal:
+            return True
+        if isinstance(signal, list) and {row["signal"], row["geo_type"]}.issubset(set(signal)):
+            return True
```

Comment:
this function gets run about 1500 times, so it might be worth worrying about speed here

Reply:
(current sirCAL running time is about a half hour)

Reply:
hmm I think it should be ok for now since the retired_signals list isn't very long. fb-survey is length 54 at the moment, but the others have ~4 signals.
```
In [20]: retired_signals = [list(range(5)) for i in range(100)]

In [21]: meta = covidcast.metadata()

In [22]: meta.shape
Out[22]: (1703, 15)

In [23]: %%time
    ...: for _, row in meta.iterrows():
    ...:     if _is_retired(row, retired_signals):
    ...:         continue
    ...: 
CPU times: user 2.26 s, sys: 0 ns, total: 2.26 s
Wall time: 2.26 s

In [24]: retired_signals = [list(range(5)) for i in range(10)]

In [25]: %%time
    ...: for _, row in meta.iterrows():
    ...:     if _is_retired(row, retired_signals):
    ...:         continue
    ...: 
CPU times: user 390 ms, sys: 7.86 ms, total: 398 ms
Wall time: 397 ms
```

I tried speeding it up with pandas apply, but that doesn't change much
```
In [72]: retired_signals = [list(range(5)) for i in range(100)]

In [73]: %time meta.apply(_is_retired, axis=1, retired_signals=retired_signals)
CPU times: user 2.27 s, sys: 0 ns, total: 2.27 s
Wall time: 2.27 s

In [74]: retired_signals = [list(range(5)) for i in range(10)]

In [75]: %time meta.apply(_is_retired, axis=1, retired_signals=retired_signals)
CPU times: user 273 ms, sys: 3.96 ms, total: 277 ms
Wall time: 276 ms

```

However, it is a bit neater and so I've put that line together with the data source filtering.

Reply:
👍